File: 3 - Searching & Sorting/Sorting/selectionSort.cpp
#include <iostream>
using namespace std;

/* Selection Sort

--> It is an algo which sorts an arr by repeatedly find the min element
--> Smaller elemenet ko lakr correct position pr rakhna h 

T.C : O(n^2)
S.C : O(1)

*/

void sort(int arr[],int n){

    for(int i=0;i<n-1;i++){
        int minIndex=i;  // i element is the smallest hai
        for(int j=i+1;j<n;j++){
            if(arr[j]<arr[minIndex]){
                minIndex=j;
            }
        }
        swap(arr[i],arr[minIndex]);
    }

    for(int i=0;i<n;i++){
        cout << arr[i] << " ";
    }
}

int main(){

    int arr[]={44,333,55,11};
    int n=4;

    sort(arr,n);
}

File: 2 - Arrays/1D Arr/selection_sort.java
public class selection_sort{

    public void selectionSort(int arr[], int n) {
		for(int i=0;i<n-1;i++){
            int minindex=i;
            for(int j=i+1;j<n;j++){
                if(arr[j]<arr[minindex]){
                    minindex=j;
                }
            }
            int temp=arr[minindex];
            arr[minindex]=arr[i];
            arr[i]=temp;
        }
        for(int i=0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
	}
    public static void main(String[] args) {
        int []arr={0,2,0,1,2,2,1,0,0};
        selection_sort s=new selection_sort();
        s.selectionSort(arr,arr.length);
    }
}